app/views/layouts/application.html.slim

```diff
@@ -46,7 +46,7 @@ html lang="#{I18n.locale}" class='no-js'
     .site-wrap
       = render 'shared/i18n_mode' if FeatureManagement.enable_i18n_mode?
       = render 'shared/usa_banner'
-      - if user_fully_authenticated?
+      - if user_fully_authenticated? && !session[:sp].present?
```

Comment:
Why only restrict this to users coming from an SP? Don't we want everyone to confirm the recovery code before being able to access the profile page? This way, if a user comes from an SP, but lets the session time out, when they start creating an account, the links will still be disabled. Because `session[:sp]` will no longer be present in that case, this conditional will return true, and will show the links to the user.